Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -133,12 +136,27 @@ @implementation _WKAttributedStringWebViewCache
     return configuration;
 }
 
++ (NSMutableArray<NSURL *> *)readOnlyAccessPaths
+{
+    static auto* readOnlyAccessPaths = [[NSMutableArray alloc] initWithCapacity:maximumReadOnlyAccessPaths];
+    return readOnlyAccessPaths;
+}
+
 + (WKWebViewConfiguration *)configuration
 {
     auto& configuration = globalConfiguration();
     if (!configuration) {
         configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
-        [configuration setProcessPool:adoptNS([[WKProcessPool alloc] init]).get()];
+
+        RetainPtr<WKProcessPool> processPool;
+        if (self.readOnlyAccessPaths.count) {
```

Comment:
Should check for the maximum count here (`maximumReadOnlyAccessPaths`) as well before setting it.

Reply:
We shouldn't ever be in a state where this is above 2, since we guard against it when ready the value from @"WKReadAccessFileURLsOption". I'll add a RELEASE_ASSERT here, since a > 2 value would indicate some trickery that should probably be a catastrophic app termination.